packages/@react-spectrum/picker/src/Picker.tsx

```diff
@@ -134,6 +134,20 @@ function Picker<T extends object>(props: SpectrumPickerProps<T>, ref: DOMRef<HTM
     }
   }, [scale, isMobile, triggerRef, state.selectedKey]);
 
+  // Make sure we only unwrap if the trigger changes otherwise we'll retrigger
+  // the observer too often
+  let resizeRef = useMemo(() => unwrapDOMRef(triggerRef), [triggerRef]);
+
+  let onResize = useCallback(() => {
+    let width = resizeRef.current.offsetWidth;
+    setButtonWidth(width);
+  }, [resizeRef, setButtonWidth]);
```

Comment:
Got a nitpick (and I'm totally fine to have this go in as is) but what do you think about moving this code a couple lines up and then reusing `onResize` in the `useLayoutEffect` above? The code inside the useLayoutEffect is basically identical so would be nice to replace it.

Reply:
I also noticed that but because I filed #1313 and I plan to revisit I wanted to keep the change focused on the bugfix. I just wanted to point out that beside the few lines above there's also `<HiddenSelect>` who uses the same unwrap-sequence, and other parts so probably this means that it could be the best thing to move that to the top?

Reply:
Yeah that seems like a good idea, I'll approve this pull then and that all can be done in a followup, thanks!